siliconcompiler/core.py

```diff
@@ -1533,12 +1541,14 @@ def check_manifest(self):
 
         #4. Check per tool parameter requirements (when tool exists)
         for step in steplist:
+
             for index in self.getkeys('flowgraph', flow, step):
                 tool = self.get('flowgraph', flow, step, index, 'tool')
+                tool = self.get('flowgraph', flow, step, index, 'task')
```

Comment:
task? Since this is also validation, we should add checks and raise error if tool and task are not set